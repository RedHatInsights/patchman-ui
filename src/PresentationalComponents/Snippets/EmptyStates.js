import {
    EmptyState,
    EmptyStateIcon,
    EmptyStateVariant,
    Title
} from '@patternfly/react-core';
import SearchIcon from '@patternfly/react-icons/dist/js/icons/search-icon';
import React from 'react';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';

export const EmptyAdvisoryList = () => (
    <EmptyState variant={EmptyStateVariant.full}>
        <EmptyStateIcon icon={SearchIcon} />
        <Title headingLevel="h5" size="lg">
            {intl.formatMessage(messages.statesNoMatchingAdvisories)}
        </Title>
    </EmptyState>
);

export const EmptyPackagesList = () => (
    <EmptyState variant={EmptyStateVariant.full}>
        <EmptyStateIcon icon={SearchIcon} />
        <Title headingLevel="h5" size="lg">
            {intl.formatMessage(messages.statesNoMatchingPackages)}
        </Title>
    </EmptyState>
);

export const EmptyCvesList = () => (
    <EmptyState variant={EmptyStateVariant.full}>
        <EmptyStateIcon icon={SearchIcon} />
        <Title headingLevel="h5" size="lg">
            {intl.formatMessage(messages.statesNoMatchingCve)}
        </Title>
    </EmptyState>
);
