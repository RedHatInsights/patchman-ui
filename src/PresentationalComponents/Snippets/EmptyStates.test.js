import { EmptyAdvisoryList, EmptyPackagesList } from './EmptyStates';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('EmptyAdvisoryList', () => {
    it('Should match the snapshot', () => {
        const wrapper = shallow(<EmptyAdvisoryList />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('EmptyPackagesList', () => {
    it('Should match the snapshot', () => {
        const wrapper = shallow(<EmptyPackagesList />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
