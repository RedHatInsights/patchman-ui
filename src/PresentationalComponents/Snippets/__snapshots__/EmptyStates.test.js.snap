// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyAdvisoryList Should match the snapshot 1`] = `
<EmptyState
  variant="full"
>
  <Title
    headingLevel="h5"
    size="lg"
  >
    No matching advisories found
  </Title>
  <EmptyStateBody>
    To continue, edit your filter settings and search again.
  </EmptyStateBody>
</EmptyState>
`;

exports[`EmptyPackagesList Should match the snapshot 1`] = `
<EmptyState
  variant="full"
>
  <EmptyStateIcon
    icon={[Function]}
  />
  <Title
    headingLevel="h5"
    size="lg"
  >
    No matching packages found
  </Title>
  <EmptyStateBody>
    To continue, edit your filter settings and search again.
  </EmptyStateBody>
</EmptyState>
`;
